File: business/src/main/java/work/beltran/discogsbrowser/business/LoginInteractor.java
package work.beltran.discogsbrowser.business;

import android.net.Uri;

import rx.Observable;

/**
 * Created by Miquel Beltran on 8/28/16
 * More on http://beltran.work
 */
public interface LoginInteractor {
    Observable<Uri> requestTokens(String apiConsumerKey, String apiConsumerSecret);

    Observable<Boolean> requisterAccessToken(Uri uri, String apiConsumerKey, String apiConsumerSecret);
}

File: business/src/main/java/work/beltran/discogsbrowser/business/LoginInteractor.kt
package work.beltran.discogsbrowser.business

import io.reactivex.Completable
import io.reactivex.Single

/**
 * Created by Miquel Beltran on 8/28/16
 * More on http://beltran.work
 */
typealias Url = String

interface LoginInteractor {
    fun requestTokens(apiConsumerKey: String, apiConsumerSecret: String): Single<Url>
    fun requisterAccessToken(redirectUrl: Url, apiConsumerKey: String, apiConsumerSecret: String): Completable
}
